mock_service/lib/types.py

```diff
@@ -28,6 +32,6 @@ class ExecutionContext(NamedTuple):
     """Context in which behaviors are executed"""
 
     calling_signal_path: str
-    pending_event_list: List[Event]
-    datapoints_cache: Dict[str, MockedDataPoint]
+    pending_event_list: List[Event] 
     delta_time: float
+    client: VSSClient = VSSClient(vdb_split[0], vdb_split[1])
```

Comment:
Should be passed as dependency injection.

Reply:
```
class ExecutionContext(NamedTuple):
    """Context in which behaviors are executed"""

    def __init__(self, vss_client:VSSClient = VSSClient(vdb_split[0], vdb_split[1])):
        self.vss_client = vss_client
        self.calling_signal_path: str
        self.pending_event_list: List[Event] 
        self.delta_time: float
       ```
so something like this?

Reply:
I would even remove the default initializer completely that way you maybe can share the same VSS client and get away without (re-)connecting? Since ExecutionContext is already a NamedTuple, you do not need the extra constructor.